### DISCUSSION TITLE
Default values

### DISCUSSION THREAD

**Discussion by redtide** [labwc/labwc-tweaks#55] [Q&A]
One thing I had in mind with my idea to use a xsd schema based configurator was, other than define and manage the structure of the types, the default values, to avoid to hard code them in the code/UI, and so avoid to keep in sync in different places (rc.xml.all, manual..).

Any idea?

**Comment by johanmalm** [comment]
My sense is that it's not worth the effort. But please take that in the context that I don't have any experience with xsd and have no understanding of how we would change labwc, labwc-tweaks, documentation or example config files if we were to adopt this.

I you feel that an xsd approach would help us more that it would hurt us, then a best next step is probably to work up a high-level case with some pros/cons and some early thoughts on implementation (changes required to labwc and tweaks). That way we can critique it and move the conversation forward.

**Reply by redtide** [reply]
I wasn't explaining the thing correctly: I don't want to use XSD in Tweaks, it was an example from what I was talking about months ago of my need to extract information like type descriptions and default values.

In this case it would be by using rc.xml.all, as the conclusion of the today IRC conversation with Consolatis.

Otherwise IDK from where to load defaults to set the entire UI widget values.
